login-utils/chsh.c

```diff
@@ -88,11 +92,45 @@ static void __attribute__((__noreturn__)) usage(void)
  */
 static void print_shells(void)
 {
+#if defined (HAVE_LIBECONF) && defined (USE_VENDORDIR)
+        size_t size = 0;
+        econf_err error;
+        char **keys = NULL;
+        econf_file *key_file = NULL;
+
+        error = econf_readDirs(&key_file,
+                               _PATH_VENDORDIR,
+                               "/etc",
+                               "shells",
+                               NULL,
+                               "", /* key only */
+                               "#" /* comment */);
+        if (error) {
+	        errx(EXIT_FAILURE,
+		     _("Cannot parse shells files: %s"),
+		     econf_errString(error));
+        }
+
+        error = econf_getKeys(key_file, NULL, &size, &keys);
+        if (error) {
+                econf_free(key_file);
+                errx(EXIT_FAILURE,
+		  _("Cannot evaluate entries in shells files: %s"),
+		  econf_errString(error));
+        }
+
```

Comment:
All this duplicates code from lib/shells.c. I guess we need more love in the area ;-)

What about adding `print_shells(FILE *out, const char *format)` to lib/shells.c, reuse econf initialization, etc.? And in long-utils/chsh.c you can use `print_shells(stdout, "%s\n")`.